### PR TITLE
Fix save booking title unescaped

### PR DESCRIPTION
**Pull request by G3root**
## What does this PR do?
the encoding of special characters in booking title bug which was closed by  #7490 still appears to be persist. this PR aims to fix that

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: packages/core/event.ts [hunk 1]
```diff
@@ -17,6 +17,9 @@ export function getEventName(eventNameObj: EventNameObjectType, forAttendeeView
       eventName: eventNameObj.eventType,
       host: eventNameObj.host,
       attendeeName: eventNameObj.attendeeName,
+      interpolation: {
```
Comment (G3root) [inline packages/core/event.ts:20]: as per https://github.com/calcom/cal.com/issues/7255#issuecomment-1442027857 . shout out to @CarinaWolli

File: packages/emails/templates/_base-email.ts [hunk 2]
```diff
@@ -40,7 +41,7 @@ export default class BaseEmail {
     const parseSubject = z.string().safeParse(payload?.subject);
     const payloadWithUnEscapedSubject = {
       ...payload,
-      ...(parseSubject.success && { subject: decodeURIComponent(parseSubject.data) }),
+      ...(parseSubject.success && { subject: decodeHTML(parseSubject.data) }),
```
[the remark this answers is not in the document]
Comment (keithwillcode) [reply packages/emails/templates/_base-email.ts:44]: Are we sure we can remove using decodeURIComponent completely? Using the playground you linked here, if I add the text `http%3A%2F%2Fwww.testing.com%2F` to the string to be decoded, decodeHTML doesn't work whereas decodeURIComponent does.

Perhaps the solution is to use both? (Note that I'm currently unfamiliar with the expected data to be passed to this function but simply an observation I made while reviewing.)

Screen Shot 2023-03-12 at 8 50 08 PM

**Review by keithwillcode — COMMENTED** [review]
See comment re: changing out function completely to use `decodeHTML`

**Comment by G3root** [comment]
@keithwillcode pushed some changes

**Review by keithwillcode — APPROVED** [review]

**Comment by G3root** [comment]
@keithwillcode some new changes. removed `decodeURIComponent` again because it throws a `malformed URI sequence`  exception if part of the string contains some `surrogate pairs` .  eg: `\uD800`   source: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Malformed_URI.

in the future we won't be affected by this problem after this change https://github.com/calcom/cal.com/issues/7255#issuecomment-1442027857 and the subject is decoded as a fallback for booking titles, which has been already saved with encoded titles . i think giving up on decoding url is to good to go for now , whats your thought ?

File: apps/web/components/booking/BookingListItem.tsx [hunk 3]
```diff
@@ -224,7 +224,7 @@ function BookingListItem(booking: BookingItemProps) {
   const showRecordingsButtons =
     (booking.location === "integrations:daily" || booking?.location?.trim() === "") && isPast && isConfirmed;
 
-  const title = decodeURIComponent(booking.title);
+  const title = booking.title;
```
Comment (G3root) [inline apps/web/components/booking/BookingListItem.tsx:227]: introduced in https://github.com/calcom/cal.com/pull/7490 to decode encoded booking title, throw this error if the title has malformed URI. so removed

Screenshot 2023-03-13 at 21-16-42 Buchungen Cal com

**Comment by keithwillcode** [comment]
> @keithwillcode some new changes. removed `decodeURIComponent` again because it throws a `malformed URI sequence` exception if part of the string contains some `surrogate pairs` . eg: `\uD800` source: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Malformed_URI.
> 
> in the future we won't be affected by this problem after this change [#7255 (comment)](https://github.com/calcom/cal.com/issues/7255#issuecomment-1442027857) and the subject is decoded as a fallback for booking titles, which has been already saved with encoded titles . i think giving up on decoding url is to good to go for now , whats your thought ?

Cool as long as we aren't losing some existing functionality people rely on today, sounds good to me.

File: apps/web/package.json [hunk 4]
```diff
@@ -69,6 +69,7 @@
     "bcryptjs": "^2.4.3",
     "classnames": "^2.3.1",
     "dotenv-cli": "^6.0.0",
+    "entities": "^4.4.0",
```
Comment (zomars) [inline apps/web/package.json:72]: @G3root @keithwillcode This new dependency seems like it's not being used. Let's revert this change please 🙏🏽
